### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by excaliware**
Please verify the authentication branch

### CODE REVIEW

File: application/src/main/java/com/musicforall/web/IndexController.java [hunk 1]
```diff
@@ -2,24 +2,52 @@
...
 import org.springframework.ui.Model;
 import org.springframework.web.bind.annotation.RequestMapping;
 
+import javax.servlet.http.HttpServletRequest;
 import java.util.Date;
 
 @Controller
 public class IndexController {
+	@Autowired
+	HttpServletRequest request;
```
Comment (cat-eat-dog) [inline application/src/main/java/com/musicforall/web/IndexController.java]: why do you need autowired request instance&

File: application/src/main/java/com/musicforall/web/IndexController.java [hunk 2]
```diff
@@ -2,24 +2,52 @@
...
+		LOG.debug("");
 	}
 
-	@RequestMapping("/")
-	public String index(Model model) {
+	@RequestMapping(value = {"/", "index"})
+	public String index(Model model, CsrfToken csrf) {
+
 		model.addAttribute("date", new Date());
+		model.addAttribute("_csrf", csrf);
```
Comment (cat-eat-dog) [inline application/src/main/java/com/musicforall/web/IndexController.java]: its better to create filter for csrf token

File: application/src/main/webapp/WEB-INF/velocity/headerIncludes.vm [hunk 3]
```diff
@@ -0,0 +1,10 @@
+<meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
```
Comment (cat-eat-dog) [inline application/src/main/webapp/WEB-INF/velocity/headerIncludes.vm]: do it using macros

File: application/src/main/webapp/WEB-INF/velocity/login.vm [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+</head>
+<body>
+<h2>Spring Demo</h2>
+
+<p>
+    <a href="index" class="btn btn-default">Main Page</a>
+</p>
+
+<form action="#springUrl('/login')" method="post">
+    <table>
```
Comment (cat-eat-dog) [inline application/src/main/webapp/WEB-INF/velocity/login.vm]: dont use tables for layout, use bootstrap intead

File: application/src/main/webapp/WEB-INF/velocity/logoutForm.vm [hunk 5]
```diff
@@ -0,0 +1,4 @@
+<form action="#springUrl('/logout')" method="post">
```
Comment (cat-eat-dog) [inline application/src/main/webapp/WEB-INF/velocity/logoutForm.vm]: create macros for form and include csrf token here

File: application/src/main/java/com/musicforall/web/IndexController.java (hunk 1 above)
Comment (excaliware) [inline application/src/main/java/com/musicforall/web/IndexController.java]: I needed it to test authentication. It is not needed now indeed.

**Comment by excaliware** [comment]
Changes made.

**Comment by excaliware** [comment]
Updated to the current master branch.

**Pull request merged**